pkg/solana/logpoller/loader.go

```diff
@@ -434,7 +431,7 @@ func (p *orderedParser) sendReadySlots() error {
 	rmvIdx := make([]int, 0)
 
 	// start at the lowest block and find ready blocks
-	for idx, block := range p.blocks {
+	for idx, block := range *p.blocks {
```

Comment:
Iterating over them directly like this won't work right. The order is of the elements of a heap is well-defined, but it's not stored in the same order as the underlying elements of the backing slice (and I think some elements might be empty?)

To get the right order, you pop them one at a time. Something like this:
```
for p.blocks.Len() > 0 {
	block := heap.Pop(p.blocks)
	...
}
```
I left another comment somewhere else suggesting to replace the `Delete` method with a call to `heap.Remove`, but now that I'm looking at this part of the code I realize you shouldn't need either of those, I think `Pop` is all we need?

Reply:
Update: replace `heap.Pop` line above with `block := (*p.blocks)[0]`, to delay the actual Pop for later when you are sure it can be removed.